core/src/proof.rs

```diff
@@ -123,10 +129,10 @@ pub enum PathProofVerificationError {
 /// either not a leaf or contains a value for a different key.
 #[derive(Clone)]
 pub struct VerifiedPathProof {
```

Comment:
making these fields public ruins the type-safety that we had here. better to find another way.